src/extension_steps.md

```diff
@@ -0,0 +1,33 @@
+# Extension
+
+## Requirements
+- My solution should support the ability to define specific letters as double and triple score.
+  - `do{u}ble` for a double letter score
+  - `tr[i]ple` for a triple letter score
+- Similar to above, my solution should support the ability to define double and triple word scores.
+  - `{double}` for a double word score
+  - `[triple]` for a triple word score
+    
+## Edge cases
+- `do}g`, `do{g`, `do]g` and `do[g`    should return 0
+- `{[d]og}` should return 18 
+
+## Implementations
+### 1
+00 Create a multiplier variable, which we will multiply the characters' score by.
+00 Validate the string for lone `]` or `}`, and also for unacceptable characters such as `|`.
+1. Start looping through the string as normal.
+2. If a `[` is encoutered, double the multiplier. When a `]` is encountered, halve the multiplier.
+3. Similarly, if a `{` is encoutered, triple the multiplier. When a `}` is encountered, third the multiplier.
+
+### 2
+Using the match function:
+00 Validate the string for lone `]` or `}`, and also for unacceptable characters such as `|`. If invalid, return 0.
+  - If match() return null, but `string` still contains `[`, `]`, `{` or `}` => invalidate
+1. Using the match() function, find the first instance of `{}` or `[]`.
+  - find the score for the string inside of `{}` or `[]` including the relevant multiplier.
+  - Remove the matched string (including `{}` or `[]`) from the input string.
+  - Repeat until match() returns `null`.
+2. Loop through the remaining string and calculate its score (no multiplier).
```

Comment:
This is sublime. Looking at this section gives a clear idea of what's to expect in the code that follows. Nice! Well done!